packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -2515,7 +2530,7 @@ export class StripeHelper extends StripeHelperBase {
       invoiceDiscountAmountInCents,
       invoiceTaxAmountInCents,
       invoiceDate: new Date(invoiceDate * 1000),
-      nextInvoiceDate: new Date(nextInvoiceDate * 1000),
+      nextInvoiceDate: nextInvoiceDate && new Date(nextInvoiceDate * 1000),
```

Comment:
According to the logic above, there should always be either a `subscriptionLineItem` or an `invoiceItemLineItem`, so why do we need to existence check `nextInvoiceDate` here? What happens downstream if `nextInvoiceDate` is `undefined`?

Reply:
It is possible for `invoiceitemLineItem` to be undefined. The logic specifically looks for the invoiceitem without `proration_details.credited_items`.

That said, in all instances where `nextInvoiceDate` is required downstream, there should be a `subscriptionLineItem`. 

The additions of `invoiceitemLineItem` should only really cover for the scenario where there is no `subscriptionLineItem` which currently only happens in the scenario where a user upgrades, and then requests a subscription cancellation.

Reply:
Also just wanted to mention that I realize this change is a bit of a quick fix. I couldn't think of a way to get a clean fix in without making significant changes to the function.